### PR TITLE
Autocompleted Author Tags

### LINKED ISSUE #95: Autocomplete author tags on entry

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #95

### CODE REVIEW

**Comment by braxtonhall** [comment]
ok now i'm done for real this time

File: extension/ts/main/ui/autocompleted.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import {debounce} from "../util/debounce";
+
+interface AutocompleteOptions<T extends HTMLElement> {
+	container: T;
+	input: HTMLInputElement | HTMLTextAreaElement;
+	getMatches: (value: string, selectionStart: number) => Promise<string[]>;
+	newValue: (match: string, value: string, selectionStart: number) => string;
+}
+
+const autocompleted = <T extends HTMLElement>({container, input, getMatches, newValue}: AutocompleteOptions<T>): T => {
```
Comment (adirar111) [inline extension/ts/main/ui/autocompleted.ts:12]: woaaaaaah very cool a p i

File: extension/ts/main/ui/autocompleted.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	input.addEventListener("click", inputHandler);
+	input.addEventListener("keydown", inputHandler);
+
+	const closeAllLists = () => autocompleteList.replaceChildren();
+
+	container.addEventListener("click", (event) => event.stopPropagation());
+	document.addEventListener("click", closeAllLists);
+
+	// again proOOOBably shouldn't do this
```
Comment (adirar111) [inline extension/ts/main/ui/autocompleted.ts:50]: 🫣 issok

**Review by adirar111 — APPROVED** [review]
woo hooooo

**Pull request merged**